### ISSUE TITLE
LearnedInterpolateToPyramidal.forward() breaks torch.compile(fullgraph=True) by probing MPS availability

### ISSUE THREAD

**Issue by wagnerpatriota** [terrastackai/terratorch#1220]
## Describe the issue

`LearnedInterpolateToPyramidal.forward()` calls `torch.mps.is_available()` on every invocation. PyTorch Dynamo intentionally marks this host capability function as skipped, so compiling the neck with `torch.compile(..., fullgraph=True)` fails before backend lowering, including on systems that do not use MPS.

MPS availability is process-static and the class already queries it during construction to emit a warning. Caching that result in `__init__()` and using the cached boolean in `forward()` preserves the MPS-only `.contiguous()` behavior while making the module fullgraph-compatible.

This affects TerraTorch 1.2.11 and remains present on current `main` at `terratorch/models/necks.py:372`. Searches of open and closed issues and pull requests found no existing report for the Dynamo/fullgraph failure. Issue #859 and PR #1009 concern the original MPS non-contiguous tensor problem, not graph capture.

## To Reproduce (optional, but appreciated)

```python
import torch

from terratorch.models.necks import LearnedInterpolateToPyramidal


model = LearnedInterpolateToPyramidal([8] * 4).eval()
features = [torch.randn(1, 8, 4, 4) for _ in range(4)]
outputs = torch.compile(model, backend="eager", fullgraph=True)(features)
print([tuple(output.shape) for output in outputs])
```

This uses the eager backend deliberately: it isolates Dynamo capture and does not require CUDA, MPS, vLLM, or another compiler backend.

A verified fix is:

```diff
diff --git a/terratorch/models/necks.py b/terratorch/models/necks.py
@@
-        if torch.mps.is_available():
+        self._mps_is_available = torch.mps.is_available()
+        if self._mps_is_available:
@@
-        if torch.mps.is_available():
+        if self._mps_is_available:
```

## Screenshots or log output (optional)

```text
torch._dynamo.exc.Unsupported: Attempted to call function marked as skipped
  Explanation: Dynamo developers have intentionally marked that the function
  `is_available` in torch/mps/__init__.py should not be traced.
  Hint: Avoid calling the function `is_available`.

from user code:
  File "terratorch/models/necks.py", line 372, in forward
    if torch.mps.is_available():
```

With the cached boolean, the same program exits successfully and prints:

```text
[(1, 2, 16, 16), (1, 4, 8, 8), (1, 8, 4, 4), (1, 8, 2, 2)]
```

## Expected behavior (optional)

`LearnedInterpolateToPyramidal` should be capturable with `torch.compile(fullgraph=True)`. The MPS-specific contiguous conversion should remain enabled when MPS is available without querying a host capability from inside the compiled forward graph.

## Deployment information (optional)

- **TerraTorch version**: 1.2.11; also reproduced against current `main` source at `dec7badd16317a4f026601dc9760b4720fd076d9`
- **Installation source**: PyPI wheel and source checkout
- **PyTorch**: 2.11.0
- **Python**: 3.12.11
- **Minimal repro backend**: `eager`
- **Additional verified environment**: Linux on AWS Trainium2 through vLLM 0.24.0 fullgraph warmup
